tests/test_workers/test_tasks/test_utils.py

```diff
@@ -39,6 +40,18 @@ def test_run_cmd(mock_sub_run):
     mock_sub_run.assert_called_once()
 
 
+@mock.patch('iib.workers.tasks.utils.subprocess.run')
+def test_run_cmd_with_cmd_repr(mock_sub_run, caplog):
+    mock_sub_run.return_value = mock.Mock(returncode=0)
+
+    secret = 'top-secret'
+    with caplog.at_level(logging.DEBUG):
+        utils.run_cmd(['echo', secret], cmd_repr=['echo', '*****'])
+
+    mock_sub_run.assert_called_once()
+    assert secret not in caplog.text
```

Comment:
Please make sure the logger is enabled:
```suggestion
    assert '*****' in caplog.text
    assert secret not in caplog.text
```

In the past, I had issues where Flask was modifying the loggers as part of the test. I needed to do this ugly workaround to make it work:
```python
    for logger in ('iib.workers', 'iib.workers.tasks', 'iib.workers.tasks.utils'):
        logging.getLogger(logger).disabled = False
```

Reply:
You're right. The additional assert highlights that this test is not reliable. It works when running on its own, but the logs are empty when running the whole test suite. Flask must be interfering as you mentioned. I tried your suggestion, but that doesn't seem to do it. I also added `"iib"` to the tuple of loggers. I'll keep digging.

Reply:
Ah! The trick is to also set the level. It looks like doing this at only the more specific logger does the trick.

Reply:
@lcarva did use that for loop before the test?